components/core/src/streaming_archive/writer/utils.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef STREAMING_ARCHIVE_WRITER_UTILS_HPP
```

Comment:
Since we are refactoring this file, shall we refactor the functions to use trailing return as well?